tests/testsuite/build_script.rs

```diff
@@ -494,7 +494,7 @@ fn custom_build_env_var_rustc_linker_with_target_cfg() {
 
     // no crate type set => linker never called => build succeeds if and
     // only if build.rs succeeds, despite linker binary not existing.
-    p.cargo("build --target").arg(&target).run();
+    p.cargo("build").target(&target).run();
```

Comment:
This kind of dynamic one is something in my mind could benefit from this change, though I haven't seen other 100+ occurrences changed in this pull request.

Upon this point, maybe this refactor might not be worthy with this scale of churn on your side?
(updating 100+ occurrences of `--target` sounds tedious)